packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -160,14 +161,17 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     @required AxisDirection crossAxisDirection,
     @required ViewportOffset offset,
     double cacheExtent,
+    bool autoCache = false,
```

Comment:
Also, I am wondering if instead of a boolean this should be a double and called something like "cacheExtendFraction", that basically specifies how many viewport heights you want the cache extend to be. E.g. setting it to 2.0 would cache two pages. setting it to 0.5 would just cache half a page. That would give us a little more flexibility I think.

We would assert that you only provide either cacheExtend or cacheExtendFraction.